### ISSUE TITLE
How to access to result of tensor after inference

### ISSUE THREAD

**Issue by EarthMu** [pytorch/executorch#1239]
Hi, 
I am implementing executorch by following step.
1. Exporting resnet18 including softmax layer.
2. Implementing executor_runner.cpp to access to result of tensor after inference.
  
I expected that I could get each classes' result like [0,0,0,0.1,0.9] after inference(including softmax).
But when I try to access to each result and stdout, the outputs were like following:
```
OutputTensor 0 1: <Unknown EValue tag 1915577445>
OutputTensor 0 2: <Unknown EValue tag -284942848>
.
.
OutputTensor 0 14: None
```  
I expected that I could get each classes' probability like following.
```
OutputTensor 0 1: 0
OutputTensor 0 2: 0.5
.
.
OutputTensor 0 14: 0.25
OutputTensor 0 15: 0.25
```

### model export file
`export-model-resnet18.py`
```py
import torch
import torchvision.models as models
import torch.nn.functional as F

from torchvision.models.resnet import ResNet18_Weights

from torch._export import capture_pre_autograd_graph
from torch.export import export, ExportedProgram
import executorch.exir as exir

# ========== resnet18 + softmax layer ============
resnet18 = models.resnet18(weights=ResNet18_Weights.DEFAULT).eval()
resnet18.fc = torch.nn.Sequential(
        resnet18.fc,
        torch.nn.Softmax(dim=1)
        )
example_args = (torch.randn(1, 3, 224, 224), )
# ====================================

## export to exir
pre_autograd_aten_dialect = capture_pre_autograd_graph(resnet18, example_args)
## export to aten dialect
aten_dialect: ExportedProgram = export(pre_autograd_aten_dialect, example_args)
## export to edge
edge_program: exir.EdgeProgramManager = exir.to_edge(aten_dialect)
## export to executorch
from executorch.exir import ExecutorchBackendConfig, ExecutorchProgramManager
executorch_program: exir.ExecutorchProgramManager = edge_program.to_executorch(
    ExecutorchBackendConfig(
        passes=[],  # User-defined passes
    )
)
## save pte model
print("save pte file")
with open("exported-resnet18.pte", "wb") as file:
    file.write(executorch_program.buffer)
```
### executor_runner.cpp
```cpp
  for (int i = 0; i < outputs.size(); ++i) {
    std::cout << "Output " << i << ": " << outputs[i] << std::endl;
    printTypeName<decltype(outputs[i])>();
    for (int j = 0; j < 1001; ++j) {
      // address
      //std::cout << "OutputTensor 0 " << j << ": " << &outputs[0,j] << std::endl;
      // value
      std::cout << "OutputTensor 0 " << j << ": " << outputs[j] << std::endl;
    }
```  
### output while running executor_runner.cpp
```sh
(executorch) root@c2aef39cb16e:~/test/executorch# ./cmake-out/executor_runner --model_path ./exported-resnet18.pte --img_path test.jpg
Number of arguments: 5
Argument 0: ./cmake-out/executor_runner
Argument 1: --model_path
Argument 2: ./exported-resnet18.pte
Argument 3: --img_path
Argument 4: test.jpg
I 00:00:00.356738 executorch:executor_runner.cpp:139] Model file ./exported-resnet18.pte is loaded.
I 00:00:00.356793 executorch:executor_runner.cpp:148] Using method forward
I 00:00:00.356799 executorch:executor_runner.cpp:196] Setting up planned buffer 0, size 64348896.
I 00:00:00.406562 executorch:executor_runner.cpp:219] Method loaded.
I 00:00:00.406674 executorch:executor_runner.cpp:225] Inputs prepared.
I 00:02:09.169815 executorch:executor_runner.cpp:234] Model executed successfully.
I 00:02:09.169871 executorch:executor_runner.cpp:238] 1 outputs: 
OutputTensor 0 0: tensor(sizes=[1, 1000], [
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  ...,
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
])
OutputTensor 0 1: <Unknown EValue tag 1915577445>
OutputTensor 0 2: <Unknown EValue tag -284942848>
OutputTensor 0 3: <Unknown EValue tag 64348896>
OutputTensor 0 4: <Unknown EValue tag -284943136>
OutputTensor 0 5: <Unknown EValue tag -284942944>
OutputTensor 0 6: <Unknown EValue tag 939732227>
OutputTensor 0 7: <Unknown EValue tag -117183485>
OutputTensor 0 8: <Unknown EValue tag -754662396>
OutputTensor 0 9: <Unknown EValue tag -989481723>
OutputTensor 0 10: <Unknown EValue tag -788086778>
OutputTensor 0 11: <Unknown EValue tag 176>
OutputTensor 0 12: <Unknown EValue tag -287441008>
OutputTensor 0 13: <Unknown EValue tag -287443824>
OutputTensor 0 14: None
OutputTensor 0 15: None
OutputTensor 0 16: <Unknown EValue tag 168>
OutputTensor 0 17: <Unknown EValue tag -284941360>
OutputTensor 0 18: <Unknown EValue tag 280>
OutputTensor 0 19: <Unknown EValue tag 522>
OutputTensor 0 20: <Unknown EValue tag 683>
OutputTensor 0 21: <Unknown EValue tag 925>
OutputTensor 0 22: <Unknown EValue tag 1164>
```

Thanks,

**Comment by dbort** [comment]
Could you talk a little more about what you're trying to do? This code seems to walk past the end of the outputs list, so the "OutputTensor i j: <Unknown Evalue tag NNN>" messages probably happen because they cast random memory to EValues:
```
  for (int i = 0; i < outputs.size(); ++i) {
    std::cout << "Output " << i << ": " << outputs[i] << std::endl;
    printTypeName<decltype(outputs[i])>();
    for (int j = 0; j < 1001; ++j) {
      std::cout << "OutputTensor 0 " << j << ": " << outputs[j] << std::endl;
    }
  }
```
i.e., `outputs.size()` seems to be 1 in this case, so only `outputs[0]` is valid. But the inner loop accesses `outputs[1]`, `outputs[2]`, etc.

One thing to watch out for is that executor_runner is more of a demo. It always passes in tensors of all `1` elements when running. So, depending on your model, you may not see typical outputs if the model expects more-complex inputs. That might explain why you're only seeing zeros here, for example.

Given that this code is trying to print ~1000 outputs, and the tensor itself has 1000 elements, maybe you're trying to print the elements of the tensor? If that's the case you could look at https://github.com/pytorch/executorch/blob/main/extension/evalue_util/print_evalue.cpp#L237, which is what prints the block like
```
tensor(sizes=[1, 1000], [
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
  ...,
  0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
])
```
That code is an example of iterating over the elements of a tensor.

**Comment by EarthMu** [comment]
Thankyou @dbort 

I could print out each value of probability in Tensor after resnet18 inference.

**Comment by dbort** [comment]
I'm glad you were able to make it work!
